### PR TITLE
Add prepare ts

### PR DESCRIPTION
**Pull request by jiangxinmeng1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [x] Code Refactoring

## Which issue(s) this PR fixes:

issue #4828 

## What this PR does / why we need it:
Add prepare TS
1. Add prepare TS in TxnMVCCNode
2. Replay 1pc txn cmd
3. Add prepare TS in NeedWaitCommit

### CODE REVIEW

File: pkg/vm/engine/tae/catalog/replay.go [hunk 1]
```diff
@@ -50,7 +52,7 @@ func (replayer *Replayer) ReplayerHandle(group uint32, commitId uint64, payload
 	checkpoint.MaxTS = e.MaxTS
 	checkpoint.LSN = e.MaxIndex.LSN
 	for _, cmd := range e.Entries {
-		replayer.catalog.ReplayCmd(cmd, replayer.dataFactory, nil, nil, replayer.cache)
+		replayer.catalog.ReplayCmd(cmd, replayer.dataFactory, nil, nil, replayer.cache, txnif.CmdSnapshot, types.TS{})
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/catalog/replay.go:55]: The definition of ReplayCmd and other replay APIs is very hard to understand.

File: pkg/vm/engine/tae/catalog/tablemvccnode.go [hunk 2]
```diff
@@ -80,33 +81,28 @@ func (e *TableMVCCNode) ApplyDelete() (err error) {
...
 }
 
-func (e *TableMVCCNode) Prepare2PCPrepare() (err error) {
-	var ts types.TS
-	ts, err = e.TxnMVCCNode.Prepare2PCPrepare()
+func (e *TableMVCCNode) ApplyCommit(index *wal.Index) (err error) {
+	var commitTS types.TS
+	if e.Deleted && e.CreatedAt.IsEmpty() {
+		commitTS, err = e.TxnMVCCNode.ApplyCommit(index, false)
+	} else {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/catalog/tablemvccnode.go:88]: Same problem. It should not set the node as deleted.

File: pkg/vm/engine/tae/db/replay.go [hunk 3]
```diff
@@ -225,41 +225,48 @@ func (replayer *Replayer) OnTimeStamp(ts types.TS) {
 	}
 }
 
-func (replayer *Replayer) OnReplayCmd(txncmd txnif.TxnCmd, idxCtx *wal.Index) {
+func (replayer *Replayer) OnReplayCmd(txncmd txnif.TxnCmd, idxCtx *wal.Index, cmdType txnif.CmdType, commitTS types.TS) {
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/db/replay.go:228]: Revisit this API

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]

File: pkg/vm/engine/tae/catalog/basemvccnode.go [hunk 4]
```diff
@@ -16,18 +16,43 @@ package catalog
...
+	NOpNoop NodeOp = iota
+	NOpCreate
+	NOpDelete
+	NOpCustomizedStart
+)
+
 type EntryMVCCNode struct {
 	CreatedAt, DeletedAt types.TS
-	Deleted              bool
+	NodeOp               []NodeOp
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/catalog/basemvccnode.go:36]: Don't add NodeOp

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Comment by jiangxinmeng1** [comment]
@mergifyio refresh

**Pull request merged**